src/lib/support/StringBuilder.h

```diff
@@ -89,13 +90,23 @@ class StringBuilderBase
     }
 };
 
-/// a preallocated sized string builder
-template <size_t kSize>
+/// A preallocated sized string builder
+/// According to the specification there are fields with size up to 256
+/// Default buffer size is 257 to have space for the terminating null character
+/// If the buffer size is not enough the value will be truncated
+template <size_t kSize = 257>
 class StringBuilder : public StringBuilderBase
 {
 public:
+    /// Default constructor
     StringBuilder() : StringBuilderBase(mBuffer, kSize) {}
 
+    /// Constructor for char * and length
+    StringBuilder(const char * data, size_t length) : StringBuilder() { AddFormat("%.*s", static_cast<int>(length), data); }
+
+    /// Constructor for CharSpan
+    StringBuilder(const CharSpan & span) : StringBuilder(span.data(), span.size()) {}
```

Comment:
Please make use of all new things you added in unit tests: https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/tests/TestStringBuilder.cpp exists and in this case we should have very good test coverage.